src/dattobd.c

```diff
@@ -4778,7 +4778,22 @@ static asmlinkage long mount_hook(char __user *dev_name, char __user *dir_name,
 	//get rid of the magic value if its present
 	if((real_flags & MS_MGC_MSK) == MS_MGC_VAL) real_flags &= ~MS_MGC_MSK;
 
-	LOG_DEBUG("detected block device mount: %s -> %s : %lu", dev_name, dir_name, real_flags);
+        /*
+	 * modify by zhaoxi copy userspace buff to kernel
+	 * date 2019/10/12
+	 */
+	char buff_dev_name[256],buff_dir_name[256];
+	memset(buff_dev_name,0,256);
+	memset(buff_dir_name,0,256);
+	copy_from_user(buff_dev_name,dev_name,256);
```

Comment:
Check the result for error.

Reply:
what about our defined `copy_string_from_user`?

Reply:
It would be better to avoid the strdup/free. But, is it even safe to be logging in a syscall in the first place?